worlds/pokemon_crystal/items.py

```diff
@@ -49,20 +50,65 @@ def item_const_name_to_id(const_name):
+    elif mode == "youngster":
+        weighted_pool = [["RARE_CANDY", "ESCAPE_ROPE"] * 11, ["ETHER", "ELIXER", "MAX_ETHER", "MAX_ELIXER", "MYSTERYBERRY"] * 9,
+                        ["WATER_STONE", "FIRE_STONE", "THUNDERSTONE", "LEAF_STONE", "SUN_STONE", "MOON_STONE"] * 2,
+                        ["GREAT_BALL"] * 1, ["POTION", "POKE_BALL", "REPEL"] * 12, ["SUPER_POTION", "ENERGY_ROOT", "ENERGYPOWDER", "SUPER_REPEL"] * 2,
+                        ["HYPER_POTION", "FULL_RESTORE"] * 1, ["MAX_REPEL"] * 1,
+                        ["REVIVE", "REVIVAL_HERB"] * 5 + ["MAX_REVIVE"] * 1,
+                        ["HP_UP", "PP_UP", "PROTEIN", "CARBOS", "CALCIUM", "IRON"] * 1,
+                        ["HEAL_POWDER", "BURN_HEAL", "PARLYZ_HEAL", "ICE_HEAL", "ANTIDOTE", "AWAKENING", "FULL_HEAL"] * 2]
+    elif mode == "cooltrainer":
+        weighted_pool = [["RARE_CANDY", "ESCAPE_ROPE"] * 11, ["MAX_ETHER", "MAX_ELIXER", "MYSTERYBERRY"] * 9,
+                        ["WATER_STONE", "FIRE_STONE", "THUNDERSTONE", "LEAF_STONE", "SUN_STONE", "MOON_STONE"] * 5,
+                        ["SUPER_POTION", "ENERGY_ROOT", "ENERGYPOWDER", "SUPER_REPEL"] * 1, ["NUGGET", "STAR_PIECE", "STARDUST", "PEARL", "BIG_PEARL"] * 5,
+                        ["GUARD_SPEC", "DIRE_HIT", "X_ATTACK", "X_DEFEND", "X_SPEED", "X_SPECIAL"] * 10,
+                        ["HYPER_POTION", "FULL_RESTORE", "MAX_REPEL"] * 10, ["REVIVE", "REVIVAL_HERB"] * 5 + ["MAX_REVIVE"] * 10,
+                        ["HP_UP", "PP_UP", "PROTEIN", "CARBOS", "CALCIUM", "IRON"] * 10,
+                        ["HEAL_POWDER", "BURN_HEAL", "PARLYZ_HEAL", "ICE_HEAL", "ANTIDOTE", "AWAKENING", "FULL_HEAL"] * 2]
+    elif mode == "vanilla":
+        # weights are roughly based on vanilla occurrence
+        weighted_pool = [["RARE_CANDY"] * 3, ["ETHER", "ELIXER", "MAX_ETHER", "MAX_ELIXER", "MYSTERYBERRY"] * 5,
+                        ["WATER_STONE", "FIRE_STONE", "THUNDERSTONE", "LEAF_STONE", "SUN_STONE", "MOON_STONE"] * 2,
+                        ["ESCAPE_ROPE"] * 3, ["NUGGET", "STAR_PIECE", "STARDUST", "PEARL", "BIG_PEARL"] * 2,
+                        ["POKE_BALL", "GREAT_BALL", "ULTRA_BALL"] * 5,
+                        ["POTION", "SUPER_POTION", "ENERGY_ROOT", "ENERGYPOWDER"] * 12,
+                        ["HYPER_POTION", "FULL_RESTORE"] * 2, ["REPEL", "SUPER_REPEL", "MAX_REPEL"] * 3,
+                        ["REVIVE", "REVIVAL_HERB"] * 4 + ["MAX_REVIVE"] * 2,
+                        ["HP_UP", "PP_UP", "PROTEIN", "CARBOS", "CALCIUM", "IRON"] * 5,
+                        ["GUARD_SPEC", "DIRE_HIT", "X_ATTACK", "X_DEFEND", "X_SPEED", "X_SPECIAL"] * 2,
+                        ["HEAL_POWDER", "BURN_HEAL", "PARLYZ_HEAL", "ICE_HEAL", "ANTIDOTE", "AWAKENING", "FULL_HEAL"] * 5]
+    else:
+        # oops
+        weighted_pool = [["NUGGET"] * 100]
     group = random.choice(weighted_pool)
     return random.choice(group)
```

Comment:
```suggestion
group = world.random.choice(weighted_pool)
return world.random.choice(group)
```
Can move to `random` instance on `World` here